src/main/resources/assets/js/components/AppPageComponent.jsx

```diff
@@ -0,0 +1,251 @@
+    return (
+      <TaskDetailComponent
+        fetchState={this.props.tasksFetchState}
+        taskHealthMessage={model.formatTaskHealthMessage(this.props.activeTask)}
+        hasHealth={model.hasHealth()}
+        onShowTaskList={this.showTaskList}
+        task={this.props.activeTask} />
+    );
+    /* jshint trailing:true, quotmark:true, newcap:true */
+    /* jscs:enable disallowTrailingWhitespace, validateQuoteMarks, maximumLineLength */
+  },
+
+  getAppDetails: function () {
+    var model = this.props.model;
+
+    /* jshint trailing:false, quotmark:false, newcap:false */
+    /* jscs:disable disallowTrailingWhitespace, validateQuoteMarks, maximumLineLength */
+    return (
+      <TogglableTabsComponent className="page-body page-body-no-top"
+          activeTabId={this.state.activeTabId}
+          onTabClick={this.onTabClick}
+          tabs={tabs} >
+        <TabPaneComponent id="tasks">
+          <TaskViewComponent
+            collection={model.tasks}
+            fetchState={this.props.tasksFetchState}
+            fetchTasks={this.props.fetchTasks}
+            formatTaskHealthMessage={model.formatTaskHealthMessage}
+            hasHealth={model.hasHealth()}
+            onTasksKilled={this.props.onTasksKilled}
+            onTaskDetailSelect={this.showTaskDetails} />
+        </TabPaneComponent>
+        <TabPaneComponent
+          id="configuration"
+          onActivate={this.props.fetchAppVersions} >
+          <AppVersionListComponent
+            app={model}
+            fetchAppVersions={this.props.fetchAppVersions}
+            fetchState={this.props.appVersionsFetchState}
+            onRollback={this.props.rollBackApp} />
+        </TabPaneComponent>
+      </TogglableTabsComponent>
+    );
+    /* jshint trailing:true, quotmark:true, newcap:true */
+    /* jscs:enable disallowTrailingWhitespace, validateQuoteMarks, maximumLineLength */
+  },
+
+  render: function () {
+    var content;
+    var model = this.props.model;
+    var statusClassSet = React.addons.classSet({
+      "text-warning": model.isDeploying()
+    });
+
+    if (this.state.activeViewIndex === 0) {
+      content = this.getAppDetails();
+    } else if (this.state.activeViewIndex === 1)  {
+      content = this.getTaskDetailComponent();
+    }
+
+    /* jshint trailing:false, quotmark:false, newcap:false */
+    /* jscs:disable disallowTrailingWhitespace, validateQuoteMarks, maximumLineLength */
+    return (
+      <PageComponent>
+        <AppBreadcrumbsComponent
+          activeTask={this.props.activeTask}
+          activeViewIndex={this.state.activeViewIndex}
+          model={model}
+          onDestroy={this.props.onDestroy}
+          showTaskList={this.showTaskList} />
+        <div className="page-header">
+          <span className="h3 modal-title">{model.get("id")}</span>
+          <ul className="list-inline list-inline-subtext">
+            <li>
+                <span className={statusClassSet}>
```

Comment:
Though I see that `TaskListComponent` has the function and, I guess, should be using it